src/indexing.jl

```diff
@@ -0,0 +1,67 @@
+# Code for accessing/indexing KeyedDataset components
```

Comment:
I'm not sure if we wanna add docstrings from some or all of these methods. I was thinking that a section/demo in the manual would probably be better?

Reply:
Cool, thanks. 👍🏻  I was definitely on the fence.

Reply:
Docstrings added.